src/clj/fluree/db/query/api.cljc

```diff
@@ -35,7 +35,8 @@
   [query {:keys [identity did issuer] :as override-opts}]
   (update query :opts (fn [{:keys [max-fuel meta] :as opts}]
                         ;; ensure :max-fuel key is present
-                        (-> (assoc opts :max-fuel max-fuel)
+                        (-> (merge override-opts opts)
```

Comment:
Good catch! If we want the override opts to override we'll have to swap the order of the merge:
`(merge opts override-opts)`